IntegrationTests/UITests/Script/ScriptRunnerTests.cs

```diff
@@ -52,16 +51,10 @@ public void Setup()
             _exampleScript.RunInBackground = true; // avoid any dialogs popping up
         }
 
-        [TearDown]
-        public void TearDown()
-        {
-            _uiCommands = null;
-        }
-
         [OneTimeTearDown]
         public void OneTimeTearDown()
         {
-            _referenceRepository.Dispose();
+            _referenceRepository?.Dispose();
```

Comment:
That's not the question. `TearDown` and `Dispose` should never throw exceptions -- but cleanup as much as possible.

Reply:
Agree, however if any of these methods throw they will uncover bugs. Otherwise we may be masking bugs.

Reply:
Then this should be asserted explicitely after everything has been disposed.

Reply:
The object is constructed or checked that it isn't `null` before each test (i.e. `Setup()`):
https://github.com/gitextensions/gitextensions/blob/13d730b7c7233494772166020ed8c39e9ffd6130/IntegrationTests/UITests/Script/ScriptRunnerTests.cs#L32-L42

Once all tests in the fixture are run, `OneTimeTearDown()` is executed to dispose `_referenceRepository`. At this stage it cannot be `null`, otherwise it is a bug and must be investigated and fixed.